diff --git a/firmware/autogen/sl_event_handler.c b/firmware/autogen/sl_event_handler.c
--- a/firmware/autogen/sl_event_handler.c
+++ b/firmware/autogen/sl_event_handler.c
@@ -1,5 +1,6 @@
 #include "sl_event_handler.h"

+#include "app_build_config.h"
 #include "sl_board_init.h"
 #include "sl_clock_manager.h"
 #include "sl_hfxo_manager.h"
@@ -57,21 +58,29 @@ void sli_internal_init_early(void)

 void sl_driver_init(void)
 {
+#if APP_DEBUG_BUILD
   sl_debug_swo_init();
+#endif
   sl_gpio_init();
+#if APP_DEBUG_BUILD
   sl_cos_send_config();
+#endif
 }

 void sl_service_init(void)
 {
+#if APP_DEBUG_BUILD
   sl_board_configure_vcom();
+#endif
   sl_hfxo_manager_init();
   sl_mbedtls_init();
   psa_crypto_init();
   sl_se_init();
+#if APP_DEBUG_BUILD
   sl_iostream_init_instances_stage_1();
   sl_iostream_init_instances_stage_2();
   sl_cli_instances_init();
+#endif
   sl_token_manager_init();
 }

@@ -97,7 +106,9 @@ void sli_platform_process_action(void)

 void sli_service_process_action(void)
 {
+#if APP_DEBUG_BUILD
   sl_cli_instances_tick();
+#endif
 }

 void sli_stack_process_action(void)
@@ -112,11 +123,15 @@ void sli_internal_app_process_action(void)

 void sl_iostream_init_instances_stage_1(void)
 {
+#if APP_DEBUG_BUILD
   sl_iostream_debug_init();
   sl_iostream_eusart_init_instances();
+#endif
 }

 void sl_iostream_init_instances_stage_2(void)
 {
+#if APP_DEBUG_BUILD
   sl_iostream_set_console_instance();
+#endif
 }
diff --git a/firmware/autogen/sl_power_manager_handler.c b/firmware/autogen/sl_power_manager_handler.c
--- a/firmware/autogen/sl_power_manager_handler.c
+++ b/firmware/autogen/sl_power_manager_handler.c
@@ -1,4 +1,5 @@
 #include <stdbool.h>
+#include "app_build_config.h"
 #include "sl_core.h"
 #include "sl_power_manager.h"
 #include "sl_sleeptimer.h"
@@ -59,9 +60,11 @@ __WEAK sl_power_manager_on_isr_exit_t app_sleep_on_isr_exit(void)
 bool sl_power_manager_is_ok_to_sleep(void)
 {
   bool ok_to_sleep = true;
+#if APP_DEBUG_BUILD
   if (sl_cli_instances_is_ok_to_sleep() == false) {
     ok_to_sleep = false;
   }
+#endif
   if (sli_zigbee_app_framework_is_ok_to_sleep() == false) {
     ok_to_sleep = false;
   }
@@ -93,12 +96,14 @@ bool sl_power_manager_sleep_on_isr_exit(void)
   // Most of the time we want to get back to sleep until the next event occurs.
   sleep = sl_power_manager_is_latest_wakeup_internal();

+#if APP_DEBUG_BUILD
   answer = sl_iostream_eusart_vcom_sleep_on_isr_exit();
   if (answer == SL_POWER_MANAGER_WAKEUP) {
     force_wakeup = true;
   } else if (answer == SL_POWER_MANAGER_SLEEP) {
     sleep = true;
   }
+#endif

   answer = sl_legacy_hal_sleep_on_isr_exit();
   if (answer == SL_POWER_MANAGER_WAKEUP) {
